### ISSUE TITLE
PUT with Content-Type other than application/json succeeds

### ISSUE THREAD

**Issue by sethrosetter** [couchbase/couchbase-lite-java-core#1110]
Using LiteServ with the following dependencies:
```
compile 'com.couchbase.lite:couchbase-lite-android:1.2.1-2'
compile 'com.couchbase.lite:couchbase-lite-java-javascript:1.2.1-2'
compile 'com.couchbase.lite:couchbase-lite-java-listener:1.2.1-2'
```

PUT with a Content-Type other than application/json succeeds on Listener db endpoint. I would expect the PUT to fail. Detected by this failing test https://github.com/couchbaselabs/sync-gateway-tests/blob/master/tests/cbl-simple-requests.js#L124-L156

```
curl -X PUT http://127.0.0.1:5984/cblite-test/test_doc_two -d '{"count":1}' -H "Content-Type: text/html"
```
Expected:
```
{
    "status" : 406,
    "error" : "not_acceptable"
 }
```

Actual:
```
{"rev":"1-9483947665d3ac2e389c6c7a14848f82","id":"test_doc_two","ok":true}
```

**Comment by zgramana** [comment]
@hideki please cherry pick this into `releases/1.2.1`

**Comment by hideki** [comment]
@zgramana
-  previous commit has issue.  Currently I sent new PR for this issue.
- cherry pick requires us to rebuid and test CBL android/java and also rebuild and test phonegap app. At this point, I consider that this fix should not by in 1.2.1
